logr.go

```diff
@@ -188,6 +188,14 @@ type Logger interface {
 	// that name segments contain only letters, digits, and hyphens
 	// (see the package documentation for more information).
 	WithName(name string) Logger
+
+	// WithCallDepth returns a Logger that will offset the call-stack by the
+	// specified number of frames.  This is useful for users who have helper
+	// functions between the "useful" call site and the actual calls to Logger
+	// methods.  Implementations that log information about the call site (such
+	// as file, function, or line) would otherwise log information about the
+	// intermediate helper functions.  Successive calls to this are additive.
+	WithCallDepth(depth int) Logger
```

Comment:
We need to document how depth is counted.

"With 0 depth, the default, the source code location of the direct caller of the logging functions is recorded. A non-zero depth skips that number of additional call stacks."

Or something like that...